File: Algorithms/019. Remove Nth Node From End of List/Solution.py
#coding=utf-8
__author__ = 'xuxuan'
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        array=[]
        while head:
            array.append(head)
            head=head.next
        if n==1:
            if len(array)==1:
                return None
            else:
                array[-2].next=None
        else:
            array[-n].val=array[-n].next.val
            array[-n].next=array[-n].next.next
        return array[0]

File: Algorithms/019. Remove Nth Node From End of List/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        List<ListNode> list=new ArrayList<ListNode>();
        list=addlist(head, list);
        if (list.size()==n) {
            return head.next;
        }
        list.get(list.size()-n-1).next=n==1?null:list.get(list.size()-n+1);
        return head;
    }
    public  List<ListNode> addlist(ListNode head, List<ListNode> list){
        if (head!=null) {
            list.add(head);
        }else {
            return list;
        }
        return addlist(head.next, list);
    }
}